lava/magma/core/model/py/ports.py

```diff
@@ -51,27 +57,74 @@ def peek(self) -> np.ndarray:
 
 
 class PyInPortVectorSparse(PyInPort):
+    """Python implementation of Vector Sparse InPort
+    """
+    # sparse port recieves twice, for val and index.
+    # if not sparse port we need to convert to dense,
+    # vice-versa, possible scipy or numpy tools for conversion
     def recv(self) -> ty.Tuple[np.ndarray, np.ndarray]:
-        pass
+        # Draf Impl. Receives Vector from Sparse
+        #
+        # if not from PyOutPortVectorSparse we need to
+        # process the data received
+        if self._csp_port:
+            return (self._csp_port.recv(),
+                    self._csp_port.recv())
+        else:
+            return (np.zeros(self._shape, self._d_type),
+                    np.zeros(self._shape, self._d_type))
```

Comment:
In the case that PyInPortVectorSparse receives data from anything other than PyOutPortVectorSparse we need to understand what is received and how to process it. Should the compiler expose that information?

@awintel

Reply:
We need to actually implement the messaging protocol described in the powerpoint slides. Each send/recv function should be compatible with any other recv/send function.

The message header should identify from what type of class it was send from. Perhaps, given the knowledge what Port connects to what other port, it could provide further information to the port classes for them to interpret the received messages.